### ISSUE TITLE
Export Notes to PDF/Word

### ISSUE THREAD

**Issue by FelixMalfait** [twentyhq/twenty#8439]
Our friends at [BlockNote](https://github.com/TypeCellOS/BlockNote) added new features: multi-column layout + export to PDF/Word. That's something we could quickly add in the new option menu in the future.

https://github.com/TypeCellOS/BlockNote/releases/tag/v0.19.0

**Comment by Bonapara** [comment]
Export would be a great feature! Multi-column layout might be a bit cramped in the side panel?

**Comment by harshit078** [comment]
@Bonapara Should't we add the `Export to Pdf` in actions in RIghtDrawer ?

**Comment by FelixMalfait** [comment]
@harshit078 yes indeed
<img width="240" alt="Screenshot 2024-11-10 at 09 39 47" src="https://github.com/user-attachments/assets/8563ebc9-b927-4f27-b41e-e324e8bcdd45">

**Comment by emshyu** [comment]
Hi @FelixMalfait! Could I be assigned to this issue?

**Comment by FelixMalfait** [comment]
@emshyu sure thank you! (Let's focus on Export only)

**Comment by samyakpiya** [comment]
Hi @FelixMalfait!

I notice this issue was assigned to @emshyu two weeks ago, but I don't see any activity yet. If there hasn't been any progress, I'd like to take on implementing the PDF/Word export functionality. I can start working on this right away.

Let me know if that works!

**Comment by samyakpiya** [comment]
Hi @FelixMalfait,

The implementation for this might actually be more complex than it seems. Here’s an overview of the current status and some of the complexities I ran into during implementation: 

- [x] RightDrawerFooter Action Menu
The new export button needs to go in the RightDrawerActionMenuDropdown, but there wasn’t an existing action in the dropdown to reference. Adding it required creating a fresh entry point, navigating several nested components and effect setters, and wiring them all together so the button only appeared for note/task objects.

I’ve implemented some of the initial functionality but wanted to share these findings for anyone who wishes to continue. These are the primary challenges I encountered while working on the export-to-PDF feature:

- [ ] Separate Export Button for Single Notes
For individual note objects, the export action needs to appear in a different menu (the command menu). The logic within CommandMenu is distinct from the ActionMenu for RightDrawer, so you will have to introduce a second, dedicated action specific to notes. In practice, this means juggling two separate paths for where the export button should appear. Or, you could also figure out a way to filter and pass CommandMenu actions to the ActionMenu for RightDrawer.

- [ ] Implementing the PDF Export Logic
BlockNote’s documentation suggests using the editor instance for the PDF export. You will need to make this instance available to the Export button's onClick. However, simply storing the editor in a Recoil atom doesn’t work (“read-only" error): it causes immutability and type issues because the editor contains mutable internal state. Attempting to store just its configuration will raise type mismatches (especially around custom “file” blocks). Refactoring to handle these types consistently can be time-consuming and requires a careful approach.

- [ ] Dependency Updates
Besides installing some [new packages](https://www.blocknotejs.org/docs/editor-api/export-to-pdf), the existing BlockNote version has to be updated to implement the new PDF exporter package. However, some of our current methods either changed or no longer exist in the newer release, so there are additional type errors and broken functionality to address when upgrading.

**Comment by FelixMalfait** [comment]
Hey thanks, to add actions to the menu items we want to setup an inheritance system on `DEFAULT_SINGLE_RECORD_ACTIONS_CONFIG_V2`

You can look at how it's done with `BASE_RECORD_LAYOUT` / `OBJECT_SPECIFIC_LAYOUTS`, I think we could introduce a similar pattern.
```
  const recordLayout: RecordLayout = {
    ...BASE_RECORD_LAYOUT,
    ...(OBJECT_SPECIFIC_LAYOUTS[targetObjectNameSingular] || {}),
    tabs: {
      ...BASE_RECORD_LAYOUT.tabs,
      ...(OBJECT_SPECIFIC_LAYOUTS[targetObjectNameSingular]?.tabs || {}),
    },
  };

```

Agree it's bigger than it looks. Do you want to try it?
cc @bosiraphael fyi


> Dependency Updates
Would you mind giving it a try in a dedicated PR? Or are you giving up on this task?

Thanks a lot!

**Comment by samyakpiya** [comment]
Thank you for clarifying!

Previously, the Actions button in the RightDrawer did nothing (it had no actions) and I assumed that was intended behavior. I just noticed that a fix has been implemented in #9241. 

Based on your feedback, I'll create a new, dedicated PR using the suggested inheritance pattern in DEFAULT_SINGLE_RECORD_ACTIONS_CONFIG_V2. Let me know if there's anything else I should keep in mind.
